env.go

```diff
@@ -19,64 +25,191 @@ type EnvVar struct {
+		def := value.Default
+		if def == "" {
+			def = r.imported[key]
 		}
+		value.Default = def
+		m[key] = value
+	}
+	return m
+}
 
-		if v, ok := ImportEnvVars[key]; ok {
-			return v
-		}
+// Environ represents the environment with configuration
+type Environ struct {
+	// env specifies the builder's configuration from environment
+	env map[string]EnvVar
+	// imported optionally specifies environment overrides
+	imported map[string]string
+	importer EnvImporterFunc
+	once     sync.Once
+}
 
-		return v.Default
+// ImportEnvFromMakefile invokes `make` to generate configuration for this mage script.
+// The makefile target is assumed to be named `magnet-vars`.
+// Assumes the makefile is named `Makefile`
+//
+// The script outputs a set of environment variables prefixed with `MAGNET_` which
+// are used as default values for the configuration variables defined by the script.
+// Any errors are ignored since this is a best-effort operation.
+func ImportEnvFromMakefile() (env map[string]string) {
+	cmd := exec.Command("make", "magnet-vars")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil
 	}
-
-	panic("Requested environment variable hasn't been registered")
+	env, _ = ImportEnvFromReader(bytes.NewReader(out))
+	return env
 }
 
-var importOnce sync.Once
+// ImportEnvFromReader consumes configuration for this mage script from the specified reader.
+// Expects the reader to produce a list of environment variables as key=value pairs with a single
+// variable per line.
+// Only the environment variables prefixed with `MAGNET_` are considered which
+// are used as default values for the configuration variables defined by the script itself.
+func ImportEnvFromReader(r io.Reader) (env map[string]string, err error) {
+	env = make(map[string]string)
+
+	s := bufio.NewScanner(r)
+	for s.Scan() {
+		line := s.Text()
+		if strings.TrimSpace(line) == "" {
+			continue
+		}
+		cols := strings.SplitN(line, "=", 2)
+		if len(cols) != 2 || !strings.HasPrefix(cols[0], "MAGNET_") {
+			log.Printf("Skip line that does not look like magnet envar: %q\n", line)
+			continue
+		}
+		key, value := strings.TrimPrefix(cols[0], "MAGNET_"), cols[1]
+		env[key] = value
+	}
+	if s.Err() != nil {
+		return nil, trace.Wrap(s.Err())
+	}
+	return env, nil
+}
 
-func importMakeEnv() {
+// env represents the default configuration environment
+var env = newEnviron(ImportEnvFromMakefile)
 
-	importOnce.Do(func() {
-		out, err := Output(context.TODO(), "make", "magnet-vars")
-		if err != nil {
-			// suppress any errors
-			return
-		}
+func newEnviron(importer EnvImporterFunc) *Environ {
+	return &Environ{
+		importer: importer,
+		env:      make(map[string]EnvVar),
+	}
+}
 
-		lines := strings.Split(out, "\n")
-		for _, line := range lines {
-			cols := strings.SplitN(line, "=", 2)
-			ImportEnvVars[cols[0]] = cols[1]
-		}
+func (r *Environ) importOnce() {
+	r.once.Do(func() {
+		r.imported = r.importer()
 	})
 }
+
+var debianFrontend = E(EnvVar{
```

Comment:
Reintroduced the automatic configuration for progress using `DEBIAN_FRONTEND` unless explicitly configured and `XDG_CACHE_HOME` for cache location unless explicitly overridden with configuration.
The only downside now is that these I will not be able to infer from a custom configuration source but I'm fine with them being just in environment.